daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -140,17 +144,21 @@ func New(info logger.Info) (logger.Logger, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	containerStream := &logStream{
 		logStreamName:    logStreamName,
 		logGroupName:     logGroupName,
 		logCreateGroup:   logCreateGroup,
+		logNonBlocking:   logNonBlocking,
 		multilinePattern: multilinePattern,
 		client:           client,
 		messages:         make(chan *logger.Message, 4096),
 	}
-	err = containerStream.create()
-	if err != nil {
-		return nil, err
+
+	if !logNonBlocking {
```

Comment:
You could create it here. The side effect would be that a non-blocking container would spend a significant amount of time  blocking on startup in the case where this api call fails. My thinking was that a user who explicitly specified non-blocking wouldn't want to delay startup significantly on timeouts to a log api.

Reply:
At least this could do the create in a goroutine so it's not blocking the first log message?

Reply:
I mean, it could block (or just queue the message) like it does now in the case that the connection isn't available yet. Anyway, this is an optimization and not necessarily needed for this change.

Reply:
Actually, since the connection doesn't need to be available anyway to just collect the log, it would probably be best to handle the connection completely asynchronously for the non-blocking mode.

Reply:
So maybe worth it to make that change here.
The connection is needed on batch send, not on Log.